core/dbt/parser/schemas.py

```diff
@@ -885,11 +888,20 @@ def parse_patch(self, block: TargetBlock[NodeTarget], refs: ParserRef) -> None:
             # Node might be disabled. Following call returns list of matching disabled nodes
             found_nodes = self.manifest.disabled_lookup.find(patch.name, patch.package_name)
             if found_nodes:
-                # There might be multiple disabled nodes for this model
-                for node in found_nodes:
-                    # We're saving the patch_path because we need to schedule
-                    # re-application of the patch in partial parsing.
-                    node.patch_path = source_file.file_id
+                if len(found_nodes) > 1 and patch.config.get("enabled"):
+                    # There are multiple disabled nodes for this model and the schema file wants to enable one.
+                    # We have no way to know which one to enable.
+                    msg = (
+                        f"Found {len(found_nodes)} matching disabled nodes for '{patch.name}'. "
+                        "Multiple nodes for the same unique id cannot be disabled in the schema "
+                        "file. They must be disabled in `dbt_project.yml` or in the sql files."
+                    )
+                    raise ParsingException(msg)
+
+                # We're saving the patch_path because we need to schedule
+                # re-application of the patch in partial parsing.
+                node = found_nodes[0]
```

Comment:
We should patch every node in found_nodes here, so loop through with for node in found_nodes. So the code doing the append_patch might need to be factored out.